### PR TITLE
Development

### PR DESCRIPTION
**Pull request by nimeshgit**
DataHub C8Y integration
Bug fixes for ZMM and ZMK
Enable units

### CODE REVIEW

File: zmm/src/Helpers/Common/FilePathHelper.cs [hunk 1]
```diff
@@ -57,8 +57,11 @@ public static string GetFilePathById<T>(string id, List<T> records)
         public static bool IsFileNameValid(string FileName)
```
[the remark this answers is not in the document]
Comment (surbhi-software) [reply zmm/src/Helpers/Common/FilePathHelper.cs]: Completed coding.

File: zmm/src/App/App.csproj [hunk 2]
```diff
@@ -31,6 +31,7 @@
     <PackageReference Include="Quartz" Version="3.0.7" />
     <PackageReference Include="Swashbuckle.AspNetCore.Swagger" Version="5.0.0-rc3" />
     <PackageReference Include="Swashbuckle.AspNetCore.SwaggerUi" Version="5.0.0-rc3" />
+    <PackageReference Include="System.Runtime.Serialization.Json" Version="4.3.0" />
```
Comment (dheerajopensource) [inline zmm/src/App/App.csproj]: ok

File: zmm/src/App/Controllers/CodeController.cs [hunk 3]
```diff
@@ -57,7 +58,7 @@ public CodeController(IWebHostEnvironment environment, IConfiguration configurat
             this.BASEURLJUP = Configuration["JupyterServer:srvurl"];
             try
             {
-                codeResponse = CodePayload.Get();
+                codeResponse = CodePayload.Get().Where(c => c.Name.Contains("-checkpoint.ipynb") == false).ToList<CodeResponse>();
```
Comment (dheerajopensource) [inline zmm/src/App/Controllers/CodeController.cs]: ok

File: zmm/src/App/Controllers/CodeController.cs [hunk 4]
```diff
@@ -38,7 +39,7 @@ public class CodeController : Controller
         private List<CodeResponse> codeResponse;
 
         private readonly string BASEURLJUP;
-        private static string[] extensions = new[] { "py", "ipynb","r" };
+        private static string[] extensions = new[] { "py", "ipynb", "r" };
```
Comment (dheerajopensource) [inline zmm/src/App/Controllers/CodeController.cs]: clean

File: zmm/src/App/Controllers/CodeController.cs [hunk 5]
```diff
@@ -99,6 +100,8 @@ public async Task<IActionResult> Post(List<IFormFile> file)
                     }
                     existingCodeData.Clear();
                     //
+                    if (!FilePathHelper.IsFileNameValid(formFile.FileName))
```
Comment (dheerajopensource) [inline zmm/src/App/Controllers/CodeController.cs]: ok

Comment (dheerajopensource) [inline zmm/src/App/Controllers/CodeController.cs]: .

File: zmm/src/App/Controllers/CodeController.cs [hunk 6]
```diff
@@ -187,7 +190,7 @@ public IActionResult Get(bool refresh)
             {
                 CodePayload.Clear();
                 InitZmodDirectory.ScanCodeDirectory();
-                codeResponse = CodePayload.Get();
+                codeResponse = CodePayload.Get().Where(c => c.Name.Contains("-checkpoint.ipynb") == false).ToList<CodeResponse>();
```
Comment (dheerajopensource) [inline zmm/src/App/Controllers/CodeController.cs]: ok

File: zmm/src/App/Controllers/CodeController.cs [hunk 7]
```diff
@@ -287,7 +290,7 @@ public async Task<IActionResult> Post(string id, [FromBody] RequestParam fileCon
         [HttpDelete("{id}")]
         public IActionResult Delete(string id)
         {
-            if(InstancePayload.IsInstanceExists(id) && CodePayload.GetById(id).Type == "JUPYTER_NOTEBOOK")
```
Comment (dheerajopensource) [inline zmm/src/App/Controllers/CodeController.cs]: .

File: zmm/src/App/Controllers/CodeController.cs (hunk 3 above)
Comment (nimeshgit) [reply zmm/src/App/Controllers/CodeController.cs]: Ignoring *.-checkpoint.ipynb files in list.

**Pull request merged**